tests/tests_test_workflow/test_bwc_workflow/bwc_test/test_bwc_test_runner_opensearch.py

```diff
@@ -0,0 +1,135 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import unittest
+from unittest.mock import MagicMock, patch
+
+from test_workflow.bwc_test.bwc_test_runner_opensearch import BwcTestRunnerOpenSearch
+
+
+class TestBwcTestRunnerOpenSearch(unittest.TestCase):
+    def setUp(self):
+        self.args = MagicMock()
+        self.args.paths = {"opensearch": "test-path"}
+        self.args.component = "sql"
+        self.args.test_run_id = "12345"
+
+        self.test_manifest = MagicMock()
+
+    @patch("test_workflow.bwc_test.bwc_test_runner_opensearch.BwcTestStartPropertiesOpenSearch")
+    @patch("test_workflow.bwc_test.bwc_test_runner_opensearch.BwcTestSuiteOpenSearch")
+    @patch("test_workflow.bwc_test.bwc_test_runner.TestRecorder")
+    @patch("test_workflow.bwc_test.bwc_test_runner.TemporaryDirectory")
+    def test_with_bwc_test(self, mock_temp, mock_test_recorder, mock_suite, mock_properties):
+        mock_test_config = MagicMock()
+        mock_test_config.bwc_test = MagicMock()
+        self.test_manifest.components = {"sql": mock_test_config}
+
+        mock_components = MagicMock()
+        mock_component = MagicMock()
+        mock_component.name = "sql"
+        mock_components.select.return_value = [mock_component]
+        mock_bundle_manifest = MagicMock()
+        mock_bundle_manifest.components = mock_components
+
+        mock_build_manifest = MagicMock()
```

Comment:
Nit; Looks like you setup mock manifests a couple of times over, might be worthwhile to add MockBuildManifest/MockTestManifest that could condense these unit tests, or at least let them reuse the same setup logic

Reply:
Note; even if they used a 'canned' construction pattern and then you modified them afterward to fit one test or another that could be useful